.sort() cgen error when using `.sort(a.before('(').len < b.before('(').len)`

### Describe the bug

```v
struct Info {
mut:
    fields []string
}

fn main() {
    mut info := Info{}
    info.fields.sort(a.before('(').len < b.before('(').len)
    println(info.fields)
}
```

```bash
/tmp $ v b.v
==================
/tmp/v_501/b.1810673733226366145.tmp.c:1862:68: error: expected parameter declarator
VV_LOCAL_SYMBOL  int compare_18408219492981413285_string_by_before('(')_len(string* a, string* b) {
                                                                   ^
/tmp/v_501/b.1810673733226366145.tmp.c:1862:68: error: expected ')'
/tmp/v_501/b.1810673733226366145.tmp.c:1862:67: note: to match this '('
VV_LOCAL_SYMBOL  int compare_18408219492981413285_string_by_before('(')_len(string* a, string* b) {
                                                                  ^
/tmp/v_501/b.1810673733226366145.tmp.c:1862:72: error: expected function body after function declarator
VV_LOCAL_SYMBOL  int compare_18408219492981413285_string_by_before('(')_len(string* a, string* b) {
                                                                       ^
/tmp/v_501/b.1810673733226366145.tmp.c:12428:107: error: implicit declaration of function 'compare_18408219492981413285_string_by_before' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        qsort(info.fields.data, info.fields.len, info.fields.element_size, (int (*)(const void *, const void *))&compare_18408219492981413285_string_by_before('(')_len);
```


### V version

latest

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -612,7 +612,7 @@ fn (mut g Gen) gen_array_sort(node ast.CallExpr) {
 		left_name := infix_expr.left.str()
 		if left_name.len > 1 {
 			compare_fn += '_by' +
-				left_name[1..].replace_each(['.', '_', '[', '_', ']', '_', "'", '_', '(', '', ')', '', ',', ''])
+				left_name[1..].replace_each(['.', '_', '[', '_', ']', '_', "'", '_', '"', '_', '(', '', ')', '', ',', ''])
 		}
 		// is_reverse is `true` for `.sort(a > b)` and `.sort(b < a)`
 		is_reverse := (left_name.starts_with('a') && infix_expr.op == .gt)

--- a/vlib/v/gen/c/array.v
+++ b/vlib/v/gen/c/array.v
@@ -611,7 +611,8 @@ fn (mut g Gen) gen_array_sort(node ast.CallExpr) {
 		comparison_type = g.unwrap(infix_expr.left_type.set_nr_muls(0))
 		left_name := infix_expr.left.str()
 		if left_name.len > 1 {
-			compare_fn += '_by' + left_name[1..].replace_each(['.', '_', '[', '_', ']', '_'])
+			compare_fn += '_by' +
+				left_name[1..].replace_each(['.', '_', '[', '_', ']', '_', "'", '_', '(', '', ')', '', ',', ''])
 		}
 		// is_reverse is `true` for `.sort(a > b)` and `.sort(b < a)`
 		is_reverse := (left_name.starts_with('a') && infix_expr.op == .gt)

--- a/vlib/v/tests/array_sort_with_fn_call_test.v
+++ b/vlib/v/tests/array_sort_with_fn_call_test.v
@@ -1,0 +1,13 @@
+struct Info {
+mut:
+	fields []string
+}
+
+fn test_sort_with_fn_call() {
+	mut info := Info{
+		fields: ['aaa(', 'b(']
+	}
+	info.fields.sort(a.before('(').len < b.before('(').len)
+	println(info.fields)
+	assert info.fields == ['b(', 'aaa(']
+}

